integration-tests/15_achievements.ts

```diff
@@ -0,0 +1,1043 @@
+    });
+    assert.ok(student_match_regular_learning_record);
+
+    await prisma.achievement_event.deleteMany({
+        where: {
+            userId: user.userID,
+            metric: 'student_match_learned_regular',
+            relation: `match/${match.id}`,
+        },
+    });
+    await client.request(`
+        mutation StudentJoinMatchMeeting { matchMeetingJoin(matchId:${match.id}) }
+    `);
+
+    const student_match_regular_learning = await prisma.user_achievement.findFirst({
+        where: {
+            userId: user.userID,
+            group: 'student_match_regular_learning',
+            achievedAt: null,
+            recordValue: 2,
+        },
+    });
+    assert.ok(student_match_regular_learning);
+});
+
+void test('Reward pupil regular learning', async () => {
+    await adminClient.request(`mutation ResetRateLimits { _resetRateLimits }`);
+    await createPupilRegularLearningTemplate();
+
+    const { pupil, client } = await pupilTwo;
+    const user = await getUser(pupil.userID);
+
+    const match = await prisma.match.findFirst({
+        where: {
+            pupilId: pupil.pupil.id,
+            dissolved: false,
+        },
+        select: { id: true },
+    });
+    // request to generate the achievement with initial record value 1
+    await client.request(`
+        mutation PupilJoinMatchMeeting { matchMeetingJoin(matchId:${match.id}) }
+    `);
+
+    const date = new Date();
+    date.setDate(date.getDate() - 7);
+    await prisma.achievement_event.create({
+        data: {
+            userId: user.userID,
+            metric: 'pupil_match_learned_regular',
+            value: 1,
+            createdAt: date,
+            action: 'pupil_joined_match_meeting',
+            relation: `match/${match.id}`,
+        },
+    });
+    // request to set the achievements record value to 2 due to the past event generated
+    await client.request(`
+        mutation PupilJoinMatchMeeting { matchMeetingJoin(matchId:${match.id}) }
+    `);
+
+    const pupil_match_regular_learning_record = await prisma.user_achievement.findFirst({
+        where: {
+            userId: user.userID,
+            group: 'pupil_match_regular_learning',
+            achievedAt: { not: null },
+            recordValue: 2,
+        },
+    });
+    assert.ok(pupil_match_regular_learning_record);
+
+    await prisma.achievement_event.deleteMany({
+        where: {
+            userId: user.userID,
+            metric: 'pupil_match_regular_learning',
+            relation: `match/${match.id}`,
+        },
+    });
+    await client.request(`
+        mutation PupilJoinMatchMeeting { matchMeetingJoin(matchId:${match.id}) }
+    `);
+
+    const pupil_match_regular_learning = await prisma.user_achievement.findFirst({
+        where: {
+            userId: user.userID,
+            group: 'pupil_match_regular_learning',
+            achievedAt: null,
+            recordValue: 2,
+        },
+    });
+    assert.ok(pupil_match_regular_learning);
```

Comment:
What case is tested here?

Reply:
This is the same as previously explained for students